design/dsl/validation.go

```diff
@@ -234,6 +234,10 @@ func Required(names ...string) {
 		return
 	}
 
+	if at.Type != nil && at.Type.Kind() == design.ObjectKind {
+		o := at.Type.(design.Object)
+		checkMissingValidationAttribute(names, o)
```

Comment:
Please feel free to tell me I am way off the mark here. This is an attempt to check that when a Required field is defined, we also check it is a defined Attribute.